xformers/components/residual.py

```diff
@@ -86,3 +98,44 @@ def forward(self, inputs: Union[torch.Tensor, List[torch.Tensor]], *args, **kwar
 
         x = self.sublayer(*inputs, *args, **kwargs)
         return self.norm(x)
+
+
+DeepNormCoefficients = namedtuple("DeepNormCoefficients", ["alpha", "beta"])
+
+
+def get_deepnorm_coefficients(
```

Comment:
PR change 2: get the residual scaling and init scaling given the whole model, following the paper